modules/images/webp-uploads/load.php

```diff
@@ -493,6 +477,29 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 			continue;
 		}
 
+		/**
+		 * Filters whether the smaller image should be used regardless of which MIME type is preferred overall.
+		 *
+		 * This is disabled by default only because it is not part of the current WordPress core feature proposal.
+		 *
+		 * By enabling this, the plugin will compare the image file sizes and prefer the smaller file regardless of MIME
+		 * type.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param bool $prefer_smaller_image_file Whether to prefer the smaller image file.
+		 */
+		if ( apply_filters( 'webp_uploads_prefer_smaller_image_file', false ) ) {
```

Comment:
A bit of a micro-optimization, but I think it would be cleaner to apply this filter outside of any `foreach` loop here and assign the value to a variable that is then checked here. It shouldn't be necessary to re-run the filter for every size, and that could even possibly result in weird behavior since in theory the value could be different in different calls of the filter.

This is even more helpful in regards to my other comment though: We need to also add a similar file size comparison for the `full` image.

Reply:
Thanks @felixarntz this has been updated